web/src/components/Imjoy.vue

```diff
@@ -2926,52 +2926,59 @@ export default {
         throw `Window (id=${wid}) not found`
       }
     },
-    async createWindow(wconfig, _plugin) {
-      wconfig.config = wconfig.config || {}
-      wconfig.data = wconfig.data || null
-      wconfig.click2load = wconfig.click2load || false
-      wconfig.panel = wconfig.panel || null
-      if (!WINDOW_SCHEMA(wconfig)) {
-        const error = WINDOW_SCHEMA.errors(wconfig)
-        console.error("Error occured during creating window " + wconfig.name, error)
-        throw error
-      }
-      // console.log('window config', wconfig)
-      if (wconfig.type && wconfig.type.startsWith('imjoy')) {
-        // console.log('creating imjoy window', wconfig)
-        wconfig.id = 'imjoy_'+randId()
-        return this.addWindow(wconfig)
-      } else {
-        const window_config = this.registered.windows[wconfig.type]
-        // console.log(window_config)
-        if (!window_config) {
-          console.error('no plugin registered for window type: ', wconfig.type)
-          throw 'no plugin registered for window type: ', wconfig.type
-        }
-        // console.log(window_config)
-        const pconfig = wconfig //_clone(window_config)
-        //generate a new window id
-        pconfig.mode = window_config.mode
-        pconfig.id = window_config.id + '_' + randId()//window_config.name.trim().replace(/ /g, '_') + '_' + randId()
-        // console.log('creating window: ', pconfig)
-        if (pconfig.mode != 'window') {
-          throw 'Window plugin must be with mode "window"'
-        }
-        // this is a unique id for the iframe to attach
-        pconfig.iframe_container = 'plugin_window_' + pconfig.id + randId()
-        // console.log('changing id...')
-        pconfig.iframe_window = null
-        pconfig.plugin = window_config
-        pconfig.context = this.pluing_context
-        if (!pconfig.click2load) {
-          this.showPluginWindow(pconfig)
-          await this.renderWindow(pconfig)
+    createWindow(wconfig, _plugin) {
+      return new Promise((resolve, reject) => {
+        wconfig.config = wconfig.config || {}
+        wconfig.data = wconfig.data || null
+        wconfig.click2load = wconfig.click2load || false
+        wconfig.panel = wconfig.panel || null
+        if (!WINDOW_SCHEMA(wconfig)) {
+          const error = WINDOW_SCHEMA.errors(wconfig)
+          console.error("Error occured during creating window " + wconfig.name, error)
+          throw error
+        }
+        // console.log('window config', wconfig)
```

Comment:
I guess these are debug prints during testing that we don't need after merge?

Reply:
I am not sure how to handle this right now, these type of errors are meant for the plugin developer to check during developing. 

We don't have a way to separate the message for the plugin developer and the core developer yet. Perhaps, in the future we should redirect these kind of error message/ information in a log window in ImJoy. Not sure about that yet, since the dev tools are in the browser are much more powerful. 

But I agree there are still some logs I didn't remove completely.

Reply:
I just meant the commented `console.log` rows.

Reply:
Sorry, I see, that's right, I should have removed that, will do a clean up soon.
Thanks for pointing out.